src/Support/TypeToSchemaExtensions/ResponderTypeToSchema.php

```diff
@@ -0,0 +1,230 @@
+                        return [];
+                    }
+
+                    $arrayToMergeItems = $this->flattenMergeValues($arrayToMerge->items);
+
+                    $mergingArrayValuesShouldBeRequired = $item->value->templateTypes[0] instanceof LiteralBooleanType
+                        && $item->value->templateTypes[0]->value === true;
+
+                    if (!$mergingArrayValuesShouldBeRequired || $item->isOptional) {
+                        foreach ($arrayToMergeItems as $mergingItem) {
+                            $mergingItem->isOptional = true;
+                        }
+                    }
+
+                    return $arrayToMergeItems;
+                }
+
+                return [$item];
+            })
+            ->values()
+            ->all();
+    }
+
+    /**
+     * @param  Generic  $type
+     */
+    public function toResponse(Type $type)
+    {
+        $definition = $this->infer->analyzeClass($type->name);
+
+        $additional = $type->templateTypes[1 /* TAdditional */] ?? new UnknownType();
+
+        $openApiType = $this->openApiTransformer->transform($type);
+
+        if (($withArray = $definition->getMethodCallType('with')) instanceof ArrayType) {
+            $withArray->items = $this->flattenMergeValues($withArray->items);
+        }
+        if ($additional instanceof ArrayType) {
+            $additional->items = $this->flattenMergeValues($additional->items);
+        }
+
+        $shouldWrap = ($wrapKey = $type->name::$wrap ?? null) !== null
+            || $withArray instanceof ArrayType
+            || $additional instanceof ArrayType;
+        $wrapKey = $wrapKey ?: 'data';
+
+        if ($shouldWrap) {
+            $openApiType = (new \Dedoc\Scramble\Support\Generator\Types\ObjectType())
+                ->addProperty($wrapKey, $openApiType)
+                ->setRequired([$wrapKey]);
+
+            if ($withArray instanceof ArrayType) {
+                $this->mergeOpenApiObjects($openApiType, $this->openApiTransformer->transform($withArray));
+            }
+
+            if ($additional instanceof ArrayType) {
+                $this->mergeOpenApiObjects($openApiType, $this->openApiTransformer->transform($additional));
+            }
+        }
+
+        return Response::make(200)
+            ->description('`' . $this->components->uniqueSchemaName($type->name) . '`')
+            ->setContent(
+                'application/json',
+                Schema::fromType($openApiType),
+            );
+    }
+
+    public function reference(ObjectType $type)
+    {
+        return new Reference('schemas', $type->name, $this->components);
+
+        /*
+         * @todo: Allow (enforce) user to explicitly pass short and unique names for the reference.
+         * Otherwise, only class names are correctly handled for now.
+         */
+        return Reference::in('schemas')
+            ->shortName(class_basename($type->name))
+            ->uniqueName($type->name);
```

Comment:
Unreachable code since you have a return before. please comment it if you are going to leave the todo comment.